### PR TITLE
[docs] Fix navigation links

### PR DESCRIPTION
**Pull request by oliviertassinari**
Fixing navigation issues reported in https://app.ahrefs.com/site-audit/3523498/11/overview

### CODE REVIEW

**Comment by flaviendelangle** [comment]
@oliviertassinari Please add a description for context or an issue to understand the problem

If it's purely to fix a link, there is a section [here](https://mui.com/x/react-data-grid/editing/#value-parser-and-value-setter) that we can link to instead of adding back the duplicated paragraph

File: docs/data/data-grid/column-definition/column-definition.md [hunk 1]
```diff
@@ -71,9 +71,19 @@ In the following demo, a formatter is used to display the tax rate's decimal val
 
 The value generated is only used for rendering purposes.
 Filtering and sorting do not rely on the formatted value.
-Use the [`valueParser`](/x/react-data-grid/cells/#value-parser) to support filtering.
+Use the [`valueParser`](#value-parser) to support filtering.
```
Comment (oliviertassinari) [inline docs/data/data-grid/column-definition/column-definition.md:74]: 404 link

File: docs/data/data-grid/column-definition/column-definition.md [hunk 2]
```diff
@@ -71,9 +71,19 @@ In the following demo, a formatter is used to display the tax rate's decimal val
...
 
-## Rendering content
+### Value parser
```
Comment (oliviertassinari) [inline docs/data/data-grid/column-definition/column-definition.md:76]: A section that we seems to have forgotten

File: docs/data/data-grid/export/export.md [hunk 3]
```diff
@@ -110,7 +110,7 @@ The grid exports a few [selectors](/x/react-data-grid/state/#access-the-state) t
 | `gridPaginatedVisibleSortedGridRowIdsSelector` | The rows after applying the sorting rules, the filtering rules, without the collapsed rows and only for the current page (**Note**: If the pagination is disabled, it will still take the value of `page` and `pageSize`). |
 
 {{"demo": "CsvGetRowsToExport.js", "bg": "inline", "defaultCodeOpen": false}}
-When using [Row grouping](/x/react-data-grid/row-grouping), it can be useful to remove the groups from the CSV export
+When using [Row grouping](/x/react-data-grid/row-grouping/), it can be useful to remove the groups from the CSV export
```
Comment (oliviertassinari) [inline docs/data/data-grid/export/export.md:113]: 301 link

File: docs/data/data-grid/export/export.md [hunk 4]
```diff
@@ -119,7 +119,7 @@ When using [Row grouping](/x/react-data-grid/row-grouping), it can be useful to
 ### Exported cells
 
 When the value of a field is an object or a `renderCell` is provided, the CSV export might not display the value correctly.
-You can provide a [`valueFormatter`](/x/react-data-grid/cells/#value-formatter) with a string representation to be used.
+You can provide a [`valueFormatter`](/x/react-data-grid/column-definition/#value-formatter) with a string representation to be used.
```
Comment (oliviertassinari) [inline docs/data/data-grid/export/export.md:122]: 404 link

File: docs/data/data-grid/overview/overview.md [hunk 5]
```diff
@@ -106,8 +106,8 @@ You can find more details on, the [feature comparison](/x/react-data-grid/gettin
 Here are some resources you might be interested in to learn more about the grid:
 
 - The storybook used for [internal development](https://material-ui-x.netlify.app/storybook/)
-- The [source on GitHub](https://github.com/mui/mui-x/tree/master/packages/grid)
-- The [Material Design specification](https://material.io/design/components/data-tables.html) specification
```
Comment (oliviertassinari) [inline docs/data/data-grid/overview/overview.md:110]: 301 link

**Comment by oliviertassinari** [comment]
> If it's purely to fix a link, there is a section [here](https://mui.com/x/react-data-grid/editing/#value-parser-and-value-setter) that we can link to instead of adding back the duplicated paragraph

@flaviendelangle This link doesn't seem to match in this context, https://mui.com/x/react-data-grid/editing/#value-parser-and-value-setter is about editing, we are referring filtering.

File: docs/data/data-grid/column-definition/column-definition.md (hunk 2 above)
Comment (flaviendelangle) [reply docs/data/data-grid/column-definition/column-definition.md:76]: I forgot to remove the demo files 
The section is now [here](https://mui.com/x/react-data-grid/editing/#value-parser-and-value-setter)

See https://github.com/mui/mui-x/pull/4600#discussion_r856245602

**Comment by flaviendelangle** [comment]
If the problem is the filtering, then we could add a section in the filtering page 

@m4theushw

File: docs/data/data-grid/column-definition/column-definition.md (hunk 2 above)
Comment (oliviertassinari) [reply docs/data/data-grid/column-definition/column-definition.md:76]: But this section only cover row editing, isn't the value parser also used for filtering?

**Comment by oliviertassinari** [comment]
> If the problem is the filtering, then we could add a section in the filtering page

It could work as well 👍

File: docs/data/data-grid/export/export.md (hunk 4 above)
Comment (flaviendelangle) [reply docs/data/data-grid/export/export.md:123]: Yes my bad on that one

**Review by flaviendelangle — APPROVED** [review]

File: docs/data/data-grid/aggregation/aggregation.md [hunk 6]
```diff
@@ -2,7 +2,7 @@
 title: Data Grid - Aggregation
 ---
 
-# Data Grid - Aggregation [<span class="plan-premium"></span>](https://mui.com/store/items/material-ui-pro/)
+# Data Grid - Aggregation [<span class="plan-premium"></span>](https://mui.com/store/items/mui-x-premium/)
```
Comment (oliviertassinari) [inline docs/data/data-grid/aggregation/aggregation.md:5]: @joserodolfofreitas This wasn't caught by ahrefs 🙃, the don't crawl out of scopes URLs (same domain), but they do crawl external domains for 301 🤦‍♂️

**Comment by oliviertassinari** [comment]
I have created https://www.notion.so/mui-org/Detect-301-404-in-the-CI-a8a2404d62a24fccb0f10b8318150ca3, we might be able to catch these issues in the CI.

**Pull request merged**